libcontainer/configs/validate/rootless.go

```diff
@@ -0,0 +1,100 @@
+package validate
+
+import (
+	"fmt"
+	"reflect"
+	"strings"
+
+	"github.com/opencontainers/runc/libcontainer/configs"
+)
+
+type RootlessValidator struct {
+}
+
+func (v *RootlessValidator) Validate(config *configs.Config) error {
+	// There's nothing to validate.
+	if config == nil {
+		return nil
+	}
+
+	if err := v.mount(config); err != nil {
+		return err
+	}
+	// If we're running under Linux with CLONE_NEWCGROUP, we don't have to worry
+	// about any of this.
+	if !config.Namespaces.Contains(configs.NEWCGROUP) {
+		if err := v.cgroup(config); err != nil {
+			return err
+		}
+	}
+
+	// XXX: We currently can't verify the user config at all, because
+	//      configs.Config doesn't store the user-related configs. So this
+	//      has to be verified by setupUser() in init_linux.go.
+
+	return nil
+}
+
+// cgroup verifies that the user isn't trying to set any cgroup limits or paths.
+func (v *RootlessValidator) cgroup(config *configs.Config) error {
+	// Nothing set at all.
+	if config.Cgroups == nil || config.Cgroups.Resources == nil {
+		return nil
+	}
+
+	// Used for comparison.
```

Comment:
We could just fail the check if any values are set at all. No need to go check for defaults. User can create the config without setting resources or path which is simple to check.